fabric-particles-v1/src/main/java/net/fabricmc/fabric/api/particles/ParticleUtils.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.particles;
+
+import net.minecraft.particle.ParticleEffect;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.World;
+
+/**
+ * Utility class for spawning particles.
+ */
+public final class ParticleUtils {
```

Comment:
Opinion: This isn't the kind of utility that Fabric API should expose. Fabric API should be lightweight and this adds very minimal value.

Reply:
What? You mean this isn't lightweight enough!?

Reply:
The existence of this class "adds weight" so to speak. It's far less necessary than everything else in the API.

Reply:
Fair enough. I _can_ remove it. It was added more because it was part of the particle code I had before, and I felt it added value to be able to work more with vectors.